src/VirtoCommerce.SitemapsModule.Web/Controllers/Api/SitemapsModuleApiController.cs

```diff
@@ -341,36 +341,37 @@ void SendNotificationWithProgressInfo(ExportImportProgressInfo c)
                 var localTmpPath = Path.Combine(localTmpFolder, $"sitemap-{storeId}.zip");
+
+                // Create directory if not exist
                 if (!Directory.Exists(localTmpFolder))
-                {
                     Directory.CreateDirectory(localTmpFolder);
-                }
 
-                //Import first to local tmp folder because Azure blob storage doesn't support some special file access mode 
-                using (var stream = SystemFile.Open(localTmpPath, FileMode.OpenOrCreate))
+                // Remove old file if exist
+                if (SystemFile.Exists(localTmpPath))
+                    SystemFile.Delete(localTmpPath);
+
+                //Import first to local tmp folder because Azure blob storage doesn't support some special file access mode
+                using (var stream = SystemFile.Open(localTmpPath, FileMode.CreateNew))
                 {
-                    using (var zipArchive = new ZipArchive(stream, ZipArchiveMode.Create, true))
+                    using var zipArchive = new ZipArchive(stream, ZipArchiveMode.Create, true);
+
+                    // Create default sitemap.xml
+                    await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, "sitemap.xml", SendNotificationWithProgressInfo);
+
+                    var sitemapUrls = await _sitemapXmlGenerator.GetSitemapUrlsAsync(storeId);
+                    foreach (var sitemapUrl in sitemapUrls.Where(url => !string.IsNullOrEmpty(url)))
                     {
-                        await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, "sitemap.xml", SendNotificationWithProgressInfo);
-
-                        var sitemapUrls = await _sitemapXmlGenerator.GetSitemapUrlsAsync(storeId);
-                        foreach (var sitemapUrl in sitemapUrls)
-                        {
-                            if (!string.IsNullOrEmpty(sitemapUrl))
-                            {
-                                await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, sitemapUrl, SendNotificationWithProgressInfo);
-                            }
-                        }
+                        await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, sitemapUrl, SendNotificationWithProgressInfo);
                     }
                 }
+
                 //Copy export data to blob provider for get public download url
-                using (var localStream = SystemFile.Open(localTmpPath, FileMode.Open))
-                using (var blobStream = _blobStorageProvider.OpenWrite(relativeUrl))
-                {
-                    localStream.CopyTo(blobStream);
-                    notification.DownloadUrl = _blobUrlResolver.GetAbsoluteUrl(relativeUrl);
-                    notification.Description = "Sitemap download finished";
-                }
+                using var localStream = SystemFile.Open(localTmpPath, FileMode.Open);
```

Comment:
Please use explicit `using` syntax as everywhere. `using` syntax with implicit scope decrease readability (imho, because you need to check the scope) and could lead to wrong scopes. The only thing it gives - minus 2 curly bracers. So I do not think we should use it in the project because of it.